controllers/agentmachine_controller.go

```diff
@@ -459,28 +462,32 @@ func (r *AgentMachineReconciler) processBootstrapDataSecret(ctx context.Context,
 	ignitionConfig := &ignitionapi.Config{}
 	if err := json.Unmarshal(bootstrapDataSecret.Data["value"], ignitionConfig); err != nil {
 		log.WithError(err).Errorf("Failed to unmarshal user-data secret %s", *machine.Spec.Bootstrap.DataSecretName)
-		return machineConfigPool, ignitionTokenSecretRef, err
+		return machineConfigPool, ignitionTokenSecretRef, ignitionEndpointHTTPHeaders, err
 	}
 
 	if len(ignitionConfig.Ignition.Config.Merge) != 1 {
 		log.Errorf("expected one ignition source in secret %s but found %d", *machine.Spec.Bootstrap.DataSecretName, len(ignitionConfig.Ignition.Config.Merge))
-		return machineConfigPool, ignitionTokenSecretRef, errors.New("did not find one ignition source as expected")
+		return machineConfigPool, ignitionTokenSecretRef, ignitionEndpointHTTPHeaders, errors.New("did not find one ignition source as expected")
 	}
 
 	ignitionSource := ignitionConfig.Ignition.Config.Merge[0]
 	machineConfigPool = (*ignitionSource.Source)[strings.LastIndex((*ignitionSource.Source), "/")+1:]
 
 	token := ""
 	for _, header := range ignitionSource.HTTPHeaders {
-		if header.Name != "Authorization" {
-			continue
-		}
-		expectedPrefix := "Bearer "
-		if !strings.HasPrefix(*header.Value, expectedPrefix) {
-			log.Errorf("did not find expected prefix for bearer token in user-data secret %s", *machine.Spec.Bootstrap.DataSecretName)
-			return machineConfigPool, ignitionTokenSecretRef, errors.New("did not find expected prefix for bearer token")
+		switch header.Name {
+		case "Authorization":
+			expectedPrefix := "Bearer "
+			if !strings.HasPrefix(*header.Value, expectedPrefix) {
+				log.Errorf("did not find expected prefix for bearer token in user-data secret %s", *machine.Spec.Bootstrap.DataSecretName)
+				return machineConfigPool, ignitionTokenSecretRef, ignitionEndpointHTTPHeaders, errors.New("did not find expected prefix for bearer token")
+			}
+			token = (*header.Value)[len(expectedPrefix):]
+		case "TargetConfigVersionHash":
+			ignitionEndpointHTTPHeaders["TargetConfigVersionHash"] = *header.Value
+		case "NodePool":
+			ignitionEndpointHTTPHeaders["Nodepool"] = *header.Value
```

Comment:
Maybe we should make this code generic, to just pass all headers?
```
if header.Name == "Authorization" {
    // do stuff
} else {
    ignitionEndpointHTTPHeaders[header.Name] = *header.Value
}
```

Reply:
Right, yes! Thank you for catching that. Modified it as suggested